File: module-nms-util/src/main/java/taboolib/module/nms/i18n/I18n.java
package taboolib.module.nms.i18n;

import com.google.common.collect.Maps;
import taboolib.common.LifeCycle;
import taboolib.common.platform.Awake;
import taboolib.module.nms.MinecraftVersion;

import java.util.Map;

/**
 * 原版语言文件工具
 *
 * @author sky
 * @since 2020-04-04 19:33
 */
public class I18n {

    public static final Map<Integer, I18nBase> VERSION = Maps.newHashMap();

    private static I18nBase base;

    static {
        VERSION.put(7, I18n11700.INSTANCE); // 1.15
        VERSION.put(8, I18n11700.INSTANCE); // 1.16
        VERSION.put(9, I18n11700.INSTANCE); // 1.17
    }

    @Awake(LifeCycle.INIT)
    static void init() {
        base = VERSION.getOrDefault(MinecraftVersion.INSTANCE.getMajor(), I18nOrigin.INSTANCE);
        base.init();
    }

    public static I18nBase get() {
        return base;
    }
}

File: module-nms-util/src/main/kotlin/taboolib/module/nms/i18n/I18n.kt
package taboolib.module.nms.i18n

import taboolib.common.platform.Awake
import taboolib.common.platform.Platform
import taboolib.common.platform.PlatformSide
import taboolib.module.nms.MinecraftVersion.major

/**
 * 原版语言文件工具
 *
 * @author sky
 * @since 2020-04-04 19:33
 */
@Awake
@PlatformSide([Platform.BUKKIT])
object I18n {

    val version = HashMap<Int, I18nBase>()
    val instance: I18nBase

    init {
        version[7] = I18n11700 // 1.15
        version[8] = I18n11700 // 1.16
        version[9] = I18n11700 // 1.17
        instance = version.getOrDefault(major, I18nOrigin.INSTANCE)
        instance.init()
    }
}